### ISSUE TITLE
Error 500 on duplicate key found in database

### ISSUE THREAD

**Issue by nikhiljohn10** [balderdashy/sails#2976]
How to catch the error occured while db find a duplicate key in the collection. The following error is produced while i gave unique:true in the email attribute of User Model. Please help.

```
error: Sending 500 ("Server Error") response: 
 Error (E_UNKNOWN) :: Encountered an unexpected error
MongoError: E11000 duplicate key error index: sails.user.$email_1  dup key: { : "me@nikz.in" }
    at toError (/srv/www/nikz_in/subdomains/voting/node_modules/sails-mongo/node_modules/mongodb/lib/utils.js:113:11)
    at /srv/www/nikz_in/subdomains/voting/node_modules/sails-mongo/node_modules/mongodb/lib/collection.js:497:60
    at /srv/www/nikz_in/subdomains/voting/node_modules/sails-mongo/node_modules/mongodb/node_modules/mongodb-core/lib/wireprotocol/2_4_support.js:414:59
    at Callbacks.emit (/srv/www/nikz_in/subdomains/voting/node_modules/sails-mongo/node_modules/mongodb/node_modules/mongodb-core/lib/topologies/server.js:94:3)
    at null.messageHandler (/srv/www/nikz_in/subdomains/voting/node_modules/sails-mongo/node_modules/mongodb/node_modules/mongodb-core/lib/topologies/server.js:235:23)
    at Socket.<anonymous> (/srv/www/nikz_in/subdomains/voting/node_modules/sails-mongo/node_modules/mongodb/node_modules/mongodb-core/lib/connection/connection.js:259:22)
    at Socket.emit (events.js:95:17)
    at Socket.<anonymous> (_stream_readable.js:765:14)
    at Socket.emit (events.js:92:17)
    at emitReadable_ (_stream_readable.js:427:10)
    at emitReadable (_stream_readable.js:423:5)
    at readableAddChunk (_stream_readable.js:166:9)
    at Socket.Readable.push (_stream_readable.js:128:10)
    at TCP.onread (net.js:529:21)

Details:  MongoError: E11000 duplicate key error index: sails.user.$email_1  dup key: { : "me@nikz.in" }

```

**Comment by tjwebb** [comment]
I can see returning a 400 error, maybe, instead of a 500. But it is an error. Where do you want to handle it?

**Comment by nikhiljohn10** [comment]
I need to handle it in the signup function inside User Model.

```
  signup: function(input, cb){
    bcrypt.hash(input.password, salt, function(err, hash) {
      User.create({
        name: input.name,
        email: input.email,
        password: hash
      })
      .exec(cb);
    });
  }

```

**Comment by sailsbot** [comment]
Thanks for posting, @nikhiljohn10.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by LeBlaaanc** [comment]
Is it possible to have waterline (or the created.js) throw a 409 (conflict) instead somehow?

**Comment by LeBlaaanc** [comment]
I played around with using `negotiate.js` but the error Waterline is throwing is just a generic validation error so maybe it's best to just do this manually, outside of Blueprint.
